### PR TITLE
Allow shortening logparser path tag

### PR DESCRIPTION
**Pull request by kir4h**
### Required for all PRs:

- [x] Signed [CLA](https://influxdata.com/community/cla/).
- [x] Associated README.md updated.
- [x] Has appropriate unit tests.

### Description

This PR resolves https://github.com/influxdata/telegraf/issues/7290 by adding some extra configuration parameters to the `logparser` input plugin.

There is a separate commit for some lintering of the Markdown file following markdownlint rules (it can be dropped if needed).

I have included two warnings at the `Start` of the plugin when using both the prefix/suffix and the folder/extension for user awareness. They are there instead of next of the method sitting next to that logic not to trigger them for every parsing. Not sure if they are actually needed anyway because this is already explained in the documentation, but included them just in case.

### CODE REVIEW

File: plugins/inputs/logparser/logparser.go [hunk 1]
```diff
@@ -92,6 +98,19 @@ const sampleConfig = `
   ## Method used to watch for file updates.  Can be either "inotify" or "poll".
   # watch_method = "inotify"
 
+  ## Remove a prefix from the 'path' generated tag.
```
[the remark this answers is not in the document]
Comment (kir4h) [reply plugins/inputs/logparser/logparser.go:101]: Thanks! I had included it but it seems I messed up somehow while splitting `README.md` into two separate commits . I have re-added it.

**Comment by danielnelson** [comment]
I think we should make this a new processor, or include it in the `strings` processor.  We could provide typical path transformations such as dirname, basename, relpath using https://golang.org/pkg/path/filepath

**Comment by kir4h** [comment]
> I think we should make this a new processor, or include it in the `strings` processor. We could provide typical path transformations such as dirname, basename, relpath using https://golang.org/pkg/path/filepath

I'm happy with porting these to a processor if this is something other inputs could benefit from (I haven't used a lot of them so far). What I wanted to avoid is the user having to play with regex for what I think could be a common use case (at least for `logparser` plugin), but if we have a  processor with path aware settings it should be easy to configure.

Regarding `strings` vs. a new package, `strings` states `The strings plugin maps certain go string functions` so unless which change its description I would go for a path processor (it's also easier to sport for a newcomer I think). What do you think?

If we agree on this I will open a new issue and reference it in the old one.

Regarding this PR, I will close it; only doubt is: do we want to cherry pick https://github.com/influxdata/telegraf/pull/7301/commits/db79553a8b00fa9eb2c52d638395a5b8b825c5a2? It's just about some lintering on the plugin's `README.md` file.

**Comment by danielnelson** [comment]
Yeah, after more thought I think we should do it a separate `filepath` processor, though the style could be similar.  If we use the Go filepath package we can support these operations in a way that should work on Windows as well.  I think it would be good to use some of the typical unix terminology for these operations:
```toml
[[processors.filepath]]
  [[processors.filepath.basename]]
    tag = "path"
  [[processors.filepath.dirname]]
    tag = "path"
  [[processors.filepath.stem]] # name taken from python, removes one extension `foo.txt` -> `foo`
    tag = "path"
```

There are other functions in the filepath package that might be helpful, but don't need to be added initially, such as Clean, Rel, ToSlash, VolumneName.

> Regarding this PR, I will close it; only doubt is: do we want to cherry pick db79553? It's just about some lintering on the plugin's README.md file.

I think what we should do here is link over to the [grok parser](https://github.com/influxdata/telegraf/tree/master/plugins/parsers/grok), I'll take care of this.

BTW, I've been meaning to deprecate the logparser input, since the tail processor can do everything it does and there isn't a strong reason to have both.

**Comment by kir4h** [comment]
> Yeah, after more thought I think we should do it a separate `filepath` processor, though the style could be similar.

Great, https://github.com/influxdata/telegraf/issues/7342 is now created to follow up on that

> BTW, I've been meaning to deprecate the logparser input, since the tail processor can do everything it does and there isn't a strong reason to have both.

Good to know! On my first day in Telegraf I got confused about both plugins as they seemed to overlap. Chose logparser because grok was integrated, but thinking of Telegraf´s modular philosohpy the tail one makes more sense. Maybe just expanding tail´s README to show some examples of putting it together with Grok (logparser´s README was helpful to me in the begining)

Closing this PR in favor of https://github.com/influxdata/telegraf/issues/7342

**Pull request closed without merging**